src/include/misc.h

```diff
@@ -237,6 +237,25 @@
         }                                                              \
     } while (0)
 
+/*
+ * Binary search for an string key.
+ */
+#define WT_BINARY_SEARCH_STRING(key, arrayp, n, found)                 \
+    do {                                                               \
+        uint32_t __base, __indx, __limit;                              \
+        (found) = false;                                               \
+        for (__base = 0, __limit = (n); __limit != 0; __limit >>= 1) { \
+            __indx = __base + (__limit >> 1);                          \
+            if (strcmp((arrayp)[__indx], (key)) < 0) {                 \
+                __base = __indx + 1;                                   \
+                --__limit;                                             \
+            } else if (strcmp((arrayp)[__indx], (key)) == 0) {         \
+                (found) = true;                                        \
+                break;                                                 \
+            }                                                          \
+        }                                                              \
+    } while (0)
+
```

Comment:
This could do with some unit tests.

Reply:
I have done quite some testing on this. Let me know if you have anything specific in mind.

Reply:
Yes, I think some unit tests would be appropriate here - something similar in spirit to the `qsort`/`qsort_r` tests. I can help get that set up if you're not familiar with them.

Reply:
That would be great!

Reply:
Updated in [d3a6198](https://github.com/wiredtiger/wiredtiger/pull/9438/commits/d3a619866111ee59400804b1422c2f6bb32fbdcf)